alt_exit.go

```diff
@@ -51,6 +51,16 @@ func Exit(code int) {
 	os.Exit(code)
 }
 
+// handleExit runs all the Logrus atexit handlers and then terminates the program using the logger's Exit function (or os.Exit() if no exit function is set).
+func (logger *Logger) handleExit(code int) {
+	runHandlers()
+	if logger.Exit != nil {
+		logger.Exit(code)
```

Comment:
@ericklaus-wf This kinds of speaks to the odd cases where `log.Fatal` or `log.Panic` come up. They are very awkward to test. I don't really want to get into a debate about testing methodology, but it seems like this would be better covered by integration testing that would ensure that both `log.Fatal` was called and the processed exited accordingly.

Are there any unit tests in the stdlib that test `log.Fatal`? Maybe we can find inspiration here.

My main concern is that something that is meant to be a testing tool will quickly become a way to do an end around on the hooks subsystem.